installer/Helpers/RequirementsChecker.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+
+namespace Installer\Helpers;
+
+
+class RequirementsChecker
+{
+	/**
+	 * Minimum PHP Version Supported (Override is in installer.php config file).
+	 *
+	 * @var _minPhpVersion
+	 */
+	private $_minPhpVersion = '7.0.0';
+
+
+
+	/**
+	 * Check for the server requirements.
+	 *
+	 * @param  array  $requirements
+	 *
+	 * @return array
+	 */
+	public function check(array $requirements)
+	{
+		$results = [];
+		foreach ($requirements as $type => $requirement_) {
+			switch ($type) {
+				// check php requirements
+				case 'php':
+					foreach ($requirements[$type] as $requirement) {
+						$results['requirements'][$type][$requirement] = true;
+						if (!extension_loaded($requirement)) {
+							$results['requirements'][$type][$requirement]
+								= false;
+							$results['errors'] = true;
+						}
+					}
+
+					if($this->checkExec())
+					{
+						$results['requirements'][$type]['Php exec() available'] = true;
+					}
+
+					break;
+				// check apache requirements
+				case 'apache':
+					foreach ($requirements[$type] as $requirement) {
+						// if function doesn't exist we can't check apache modules
+						if (function_exists('apache_get_modules')) {
+							$results['requirements'][$type][$requirement]
+								= true;
+							if (!in_array($requirement, apache_get_modules())) {
+								$results['requirements'][$type][$requirement]
+									= false;
+								$results['errors'] = true;
+							}
+						}
+					}
+					break;
+			}
+		}
+
+		return $results;
+	}
+
+
+
+	/**
+	 * Check PHP version requirement.
+	 *
+	 * @param  string|null  $minPhpVersion
+	 *
+	 * @return array
+	 */
+	public function checkPHPversion(string $minPhpVersion = null)
+	{
+		$minVersionPhp = $minPhpVersion;
+		$currentPhpVersion = $this->getPhpVersionInfo();
+		$supported = false;
+		if ($minPhpVersion == null) {
+			$minVersionPhp = $this->getMinPhpVersion();
+		}
+		if (version_compare($currentPhpVersion['version'], $minVersionPhp)
+			>= 0
+		) {
+			$supported = true;
+		}
+		$phpStatus = [
+			'full'      => $currentPhpVersion['full'],
+			'current'   => $currentPhpVersion['version'],
+			'minimum'   => $minVersionPhp,
+			'supported' => $supported,
+		];
+
+		return $phpStatus;
+	}
+
+
+
+	/**
+	 * Check if exec is enabled. This will allow use to execute the migration
+	 *
+	 * @return bool
+	 */
+	public function checkExec()
+	{
+		$disabled = explode(',', ini_get('disable_functions'));
+
+		return !in_array('exec', $disabled);
+	}
+
+
+
+	/**
+	 * Get current Php version information.
+	 *
+	 * @return array
+	 */
+	private static function getPhpVersionInfo()
+	{
+		$currentVersionFull = PHP_VERSION;
+		preg_match("#^\d+(\.\d+)*#", $currentVersionFull, $filtered);
```

Comment:
I'm trying to understand what this does. Will it effectively strip any `-rc` strings and such, keeping just the numbers and dots?

Reply:
yep